extensions/amp-form/0.1/test-e2e/test-ssr.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const pageWidth = 800;
+const pageHeight = 600;
+
+/**
+ * Amp-form test for SSR enabled viewer.
+ */
+
+describes.endtoend('AMP form server side rendered templates', {
+  testUrl: 'http://localhost:8000/test/fixtures/e2e/amp-form.ssr.html',
+  environments: ['viewer-demo'],
```

Comment:
out of curiosity, the expectations below would fail if this ran in the single doc or shadow doc environment because the template would render something different than `Server side rendered search result`, is that right?

Reply:
correct. The server responses would differ. We can add those different environments and just have expectations based on env conditions but I think there is the potential for the test to get really messy.

Reply:
agreed, thanks!

Reply:
After talking with Will about these changes, I realized that we should test the single environment as well and modify the response so that it is similar to the response that would come from an ssr viewer. This would then test that turning SSR on and off should render the same results. I will make this change and ping again when done. Thanks